raiden/routing.py

```diff
@@ -40,47 +49,124 @@ def get_best_routes(
+            target=to_checksum_address(to_address),
+            amount=amount,
         )
+        return ("Node does not have a channel in the requested token network.", list(), None)
+
+    error_closed = 0
+    error_no_route = 0
+    error_no_capacity = 0
+    error_direct = None
+    shortest_routes: List[Neighbour] = list()
+
+    # Always use a direct channel if available:
+    # - There are no race conditions and the capacity is guaranteed to be
+    #   available.
+    # - There will be no mediation fees
+    # - The transfer will be faster
+    if to_address in all_neighbors:
+        for channel_id in token_network.partneraddresses_to_channelidentifiers[
+            Address(to_address)
+        ]:
+            channel_state = token_network.channelidentifiers_to_channels[channel_id]
+
+            # direct channels dont have fees
+            payment_with_fee_amount = PaymentWithFeeAmount(amount)
+            is_usable = channel.is_channel_usable_for_new_transfer(
+                channel_state, payment_with_fee_amount, None
+            )
 
-        for route_state in internal_routes:
-            if to_address == route_state.next_hop_address and (
-                channel_state
-                # other conditions about e.g. channel state are checked in best routes internal
-                and channel.get_distributable(
-                    sender=channel_state.our_state, receiver=channel_state.partner_state
+            if is_usable is channel.IsChannelUsable.YES:
+                direct_route = RouteState(
+                    route=[Address(from_address), Address(to_address)],
+                    forward_channel_id=channel_state.canonical_identifier.channel_identifier,
+                    estimated_fee=FeeAmount(0),
                 )
-                >= amount
-            ):
-                return (None, [route_state], None)
+                return (None, [direct_route], None)
+
+            error_direct = is_usable
+
+    for partner_address in all_neighbors:
+        for channel_id in token_network.partneraddresses_to_channelidentifiers[partner_address]:
+            channel_state = token_network.channelidentifiers_to_channels[channel_id]
+
+            if channel.get_status(channel_state) != ChannelState.STATE_OPENED:
+                error_closed += 1
+                continue
+
+            try:
+                route = networkx.shortest_path(
+                    token_network.network_graph.network, partner_address, to_address
+                )
+            except (networkx.NetworkXNoPath, networkx.NodeNotFound):
+                error_no_route += 1
+            else:
+                distributable = channel.get_distributable(
+                    channel_state.our_state, channel_state.partner_state
+                )
+
+                if distributable < amount:
+                    error_no_capacity += 1
+                else:
+                    nonrefundable = amount > channel.get_distributable(
```

Comment:
Not for this PR, but it feels like we have the same checks all over the place. We should try to pull this out to have it in one place.